### PR TITLE
support to delete remote tag from command palette.

### LINKED ISSUE #104845: Git: Add support to delete remote git tags using command palette
Be able to delete remote git tags using  this command using command palette

`git push --delete origin tagname`

currently only deletes local tags

### PR DESCRIPTION
**Pull request by Shubhamurkade**
This PR fixes #104845

Added support to delete remote tag using the following git command - 'git push --delete <remote_name> <tag_name>'.

The 'deleteRemoteTag' function in 'repository.ts' is the function which has all the logic. The following actions happen when the user selects 'Delete: Remote Tag' from command palette:

1. The user is prompted to choose a tag from the repository.
2. The user is prompted to choose a remote from the repository.
3. After that the above-mentioned git command is executed to delete the tag from the remote specified.

### CODE REVIEW

File: extensions/git/src/commands.ts [hunk 1]
```diff
@@ -1890,6 +1890,48 @@ export class CommandCenter {
...
+		const tag = await window.showQuickPick(picks, { placeHolder });
+
+		if (!tag) {
+			return;
+		}
+
+		const remotes = repository.remotes;
+
+		if (remotes.length === 0) {
+			window.showErrorMessage(localize('no remotes added', "Your repository has no remotes."));
```
Comment (joaomoreno) [inline extensions/git/src/commands.ts:1913]: We should do this validation before asking for user input.

File: extensions/git/src/commands.ts [hunk 2]
```diff
@@ -1890,6 +1890,48 @@ export class CommandCenter {
...
+			return;
+		}
+
+		const picksRemote = remotes.filter(r => r.fetchUrl !== undefined).map(r => ({ label: r.name, description: r.fetchUrl! }));
+
+		if (picksRemote.length === 0) {
+			window.showWarningMessage(localize('no remotes', "This repository has no remotes."));
```
Comment (joaomoreno) [inline extensions/git/src/commands.ts:1920]: This should be an error message, not warning. Also, the validation should happen before asking for user input.

File: extensions/git/src/commands.ts [hunk 3]
```diff
@@ -1890,6 +1890,48 @@ export class CommandCenter {
...
+		}
+
+		const remotes = repository.remotes;
+
+		if (remotes.length === 0) {
+			window.showErrorMessage(localize('no remotes added', "Your repository has no remotes."));
+			return;
+		}
+
+		const picksRemote = remotes.filter(r => r.fetchUrl !== undefined).map(r => ({ label: r.name, description: r.fetchUrl! }));
```
Comment (joaomoreno) [inline extensions/git/src/commands.ts:1917]: Why `fetchUrl`? Wouldn't `pushUrl` be a better indicator?

**Review by joaomoreno — CHANGES_REQUESTED** [review]

**Review by kadirselcuk — APPROVED** [review]

**Comment by lszomoru** [comment]
Thank you very much for your contribution. It looks like this pull request has been abandoned as the requested changes were never implemented. There is also a new pull request that adds the same functionality - https://github.com/microsoft/vscode/pull/134327. Closing this pull request.

**Pull request closed without merging**